src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -142,60 +148,46 @@ def generate_update_stmt(
-            elif param.type == "path":
-                path = path.replace(
-                    f"<{param.name}>",
-                    pydash.get(param_values, param.references[0].field),
+        collection_values: Dict[str, Row] = {collection_name: row}
+        identity_data: Dict[str, Any] = request.get_cached_identity_data()
+
+        # create the source of param values to populate the various placeholders
+        # in the path, headers, query_params, and body
+        param_values: Dict[str, Any] = {}
+        for request_param in current_request.request_params:
+            if request_param.references:
+                param_values[request_param.name] = pydash.get(
+                    collection_values, request_param.references[0].field
                 )
-            elif param.type == "body":
-                body = SaaSQueryConfig._build_request_body(
-                    path,
-                    param.name,
-                    body,
-                    param.default_value,
-                    pydash.get(param_values, param.references[0].field)
-                    if param.references
-                    else None,
-                    pydash.get(param_values, param.identity)
-                    if param.identity
-                    else None,
+            elif request_param.identity:
+                param_values[request_param.name] = pydash.get(
+                    identity_data, request_param.identity
+                )
+            elif request_param.connector_param:
+                param_values[request_param.name] = pydash.get(
+                    self.secrets, request_param.connector_param
                 )
 
-        logger.info(f"Populated request params for {current_request.path}")
-
+        # mask row values
         update_value_map: Dict[str, Any] = self.update_value_map(row, policy, request)
         update_values: Dict[str, Any] = unflatten_dict(update_value_map)
-        method: HTTPMethod = (
-            current_request.method if current_request.method else HTTPMethod.PUT
+
+        # removes outer {} wrapper from body for greater flexibility in custom body config
+        param_values["masked_object_fields"] = json.dumps(update_values)[1:-1]
+
+        # map param values to placeholders in path, headers, and query params
+        saas_request_params: SaaSRequestParams = self.map_param_values(
+            current_request, param_values
         )
-        if body:
-            # removes outer {} wrapper from body for greater flexibility in custom body config
-            body = body.replace(
-                "<masked_object_fields>", json.dumps(update_values)[1:-1]
-            )
-        return SaaSRequestParams(
-            method=method,
-            path=path,
-            params=params,
-            body=json.dumps(json.loads(body) if body else update_values),
+
+        # map body
+        body = self.assign_placeholders(current_request.body, param_values)
```

Comment:
The organization feels a bit confusing to me here, and maybe just rethinking method names will help - `map_param_values` calls `assign_placeholders` to update path, headers, and query params, but then separately you call `assign_placeholders` to adjust values in the body and then update the response from `map_param_values`. Would it be worthwhile to have `map_param_values` be able to handle the body too?

Reply:
I agree, I found a way to make it work for both `generate_query` and `generate_update_stmt`.

Reply:
looks good!